File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<int> nums)
{
    unordered_map<int, int> hash;
    
    for (auto num : nums) 
        hash[num]++;
    
    return min(hash.size(), nums.size() / 2);
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int answer = nums.length / 2;
        HashSet<Integer> hs = new HashSet<>();
        for (int num : nums) hs.add(num);
        
        if (answer >= hs.size()) return hs.size();
        return answer;
    }
}